ddtrace/appsec/_exploit_prevention/stack_traces.py

```diff
@@ -13,26 +12,40 @@
 
 
 def report_stack(
-    message: str, span: Optional[Span] = None, crop_stack: Optional[str] = None, stack_id: Optional[str] = None
-):
+    message: Optional[str] = None,
+    span: Optional[Span] = None,
+    crop_stack: Optional[str] = None,
+    stack_id: Optional[str] = None,
+    ep_or_iast: bool = True,  # True for EP, False for IAST
+) -> bool:
     """
     Report a stack trace to the current span.
     This is used to report stack traces for exploit prevention.
     Return the stack id for the reported stack trace to link it in triggers.
     """
-    if not asm_config._ep_enabled or not asm_config._ep_stack_trace_enabled:
-        return None
+    if not asm_config._ep_stack_trace_enabled:
+        # stack trace report disabled
+        return False
+    if ep_or_iast and not (asm_config._asm_enabled and asm_config._ep_enabled):
+        # exploit prevention stack trace with ep disabled
+        return False
+    if not ep_or_iast and not (asm_config._iast_enabled):
+        # iast stack trace with iast disabled
+        return False
+
     if span is None:
         span = ddtrace.tracer.current_span()
     if span is None or stack_id is None:
-        return None
+        return False
     root_span = span._local_root or span
     appsec_traces = root_span.get_struct_tag(EXPLOIT_PREVENTION.STACK_TRACES) or {}
-    exploit: List[Any] = appsec_traces.get("exploit", [])
+    namespace = "exploit" if ep_or_iast else "vulnerability"
```

Comment:
I feel like "exploit" or "vulnerability" should be constants and received by the function, instead of receiving `ep_or_iast`